.goreleaser.yaml

```diff
@@ -0,0 +1,41 @@
+# Make sure to check the documentation at https://goreleaser.com
+before:
+  hooks:
+    - go mod tidy
+builds:
+    - id: webhook
+      main: ./main.go
+      goos:
+        - linux
+      goarch:
+        - amd64
+        - arm64
+      binary: webhook
+      ldflags:
+        - -extldflags
+        - -static
+        - -s
+        - -X main.Version={{.Version}} -X main.GitCommit={{.Commit}}
```

Comment:
I guess my main concern is now we have 2 different binaries being built:
- The one with gorelease on releases
- The one with `script/build`

The binaries built in the latter are copied in the docker image and pushed to the container registry.

The binaries built in the former are pushed to the release/pre-release. I'm not sure what they're used for tbh, since afaik what's most important is the Helm chart (.tgz) for the charts repo and the docker image.

We're not trying to redo all our CI, so I'm not suggesting we drop the binaries from the release. However I wanna understand the reason to have two different ways to build the binaries, used in two different artifacts. Maintenance-wise, these two build mechanism will need to be kept in sync. I haven't used goreleaser before so I'd like to know how it benefits us for our use case.

Reply:
There's a directive around somewhere that we should avoid storing binaries in `artifacts` directories for purposes of SLSA level-3 compliance, so I didn't see a way that we could test the same binaries we push to github without loading everything into one job.

Reply:
Is there anything that would prevent replacing the goreleaser step with a `make build` step, or something like that?

Reply:
I thought we want to use goreleaser because it automates more of the release process.